extensions/amp-story/1.0/pagination-buttons.js

```diff
@@ -222,50 +208,42 @@ export class PaginationButtons {
   onCurrentPageIndexUpdate_(pageIndex) {
     const totalPages = this.storeService_.get(StateProperty.PAGE_IDS).length;
 
-    if (pageIndex === 0) {
-      this.backButton_.updateState(BackButtonStates.HIDDEN);
-    }
-
-    if (pageIndex > 0) {
-      this.backButton_.updateState(BackButtonStates.PREVIOUS_PAGE);
-    }
+    // Hide back button if no previous page.
+    this.setEnabled_(this.backButton_, pageIndex > 0);
 
     if (pageIndex < totalPages - 1) {
-      this.forwardButton_.updateState(ForwardButtonStates.NEXT_PAGE);
+      this.forwardButton_.updateState(ButtonStates.NEXT_PAGE);
     }
 
     if (pageIndex === totalPages - 1) {
       const viewer = Services.viewerForDoc(this.ampStory_.element);
       if (viewer.hasCapability('swipe')) {
-        this.forwardButton_.updateState(ForwardButtonStates.NEXT_STORY);
+        this.forwardButton_.updateState(ButtonStates.NEXT_STORY);
       } else {
-        this.forwardButton_.updateState(ForwardButtonStates.REPLAY);
+        this.forwardButton_.updateState(ButtonStates.REPLAY);
       }
     }
   }
 
+  /**
+   * Updates element based on visibility state.
+   * @param {!PaginationButton} paginationButton
+   * @param {boolean} isVisible
+   * @private
+   */
+  setEnabled_(paginationButton, isVisible) {
```

Comment:
Nit: the function may read better if the param is called `isEnabled` or `enabled` instead

Reply:
I think the different concepts can have different naming schemes and it's fine. The system UI mostly disappears when long pressing for example, so "visibility" is for the most part accurate. Since the buttons are not fully invisible when disabled (I think opacity is 0.1 or so), "enabled" for buttons still makes sense.

With the other comment ("implement the `setEnabled` directly in the `PaginationButton` class") it also makes more sense to have the signature be `setEnabled(isEnabled)` than `setEnabled(isVisible)`. WDYT?